koptional/src/main/kotlin/com/gojuno/koptional/Optional.kt

```diff
@@ -6,6 +6,8 @@ sealed class Optional<out T : Any> {
         is Some -> value
         is None -> null
     }
+
+    open operator fun component1() = toNullable()
```

Comment:
Actually, playing more with it I think we should define it as an abstract function:

```kotlin
sealed class Optional {
    abstract operator fun component1(): T?
}

// `Some` overrides it automatically, 
// also looks like we can't define it manually in either case which worries me a bit

object None : Optional<Nothing> {
    override fun component1(): Nothing? = null
}
```

That will work faster for `None` because it will just return `null` without doing `instanceof` that `toNullable()` does 

And actual behavior for `Some` will be more clear because right now it's not obvious that `toNullable()` is not actually executed due to override generated by compiler

Reply:
nice i like that